File: apps/android/app/src/main/kotlin/com/nogeass/passalarm/domain/model/AlarmPlan.kt
package com.nogeass.passalarm.domain.model

data class AlarmPlan(
    val id: Long = 0,
    val label: String = "",
    val isEnabled: Boolean = true,
    val timeHHmm: String = "07:00",
    val weekdaysMask: Int = 0b00011111, // Mon-Fri
    val repeatCount: Int = 10,
    val intervalMin: Int = 5,
    val soundId: String = "default",
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

File: apps/ios/PassAlarm/Domain/Entity/AlarmPlan.swift
import Foundation

struct AlarmPlan: Identifiable, Equatable, Sendable, Codable {
    var id: UUID
    var isEnabled: Bool
    var label: String
    var timeHHmm: String // "07:00"
    var weekdaysMask: UInt8 // Mon=1, Tue=2, Wed=4, Thu=8, Fri=16, Sat=32, Sun=64
    var repeatCount: Int
    var intervalMin: Int
    var soundId: String
    var createdAt: Date
    var updatedAt: Date

    static let `default` = AlarmPlan(
        id: UUID(),
        isEnabled: true,
        label: "",
        timeHHmm: "07:00",
        weekdaysMask: 0b00011111, // Mon-Fri
        repeatCount: 10,
        intervalMin: 5,
        soundId: "default",
        createdAt: Date(),
        updatedAt: Date()
    )
}
